src/collector/Round.cpp

```diff
@@ -221,6 +224,44 @@ void Round::step2_1_jobs(void *arg)
 
         self.m_storage->save_new_jobs(std::move(jobs), ts);
 
+        // History
+
+        time_t group_history_ts = ::time(nullptr);
+        double previous_ts = self.m_storage->get_group_history_ts();
+
+        std::vector<GroupHistoryEntry> group_history;
+
+        cursor = conn->query(config.metadata.history.db + ".history",
+                MONGO_QUERY("nodes.timestamp" << mongo::GT << previous_ts).readPref(
+                        mongo::ReadPreference_PrimaryOnly, mongo::BSONArray()));
+
+        while (cursor->more()) {
+            mongo::BSONObj obj = cursor->next();
+
+            try {
+                GroupHistoryEntry entry(obj);
+                if (!entry.empty()) {
+                    BH_LOG(app::logger(), DNET_LOG_INFO, "Loaded group history entry:\n%s",
+                            entry.to_string().c_str());
+                    group_history.emplace_back(std::move(entry));
+                }
```

Comment:
It's better to handle empty entries without exceptions, because it is not an error.